src/ConnectedMode/Migration/MigrationSettingsProvider.cs

```diff
@@ -39,15 +36,12 @@ internal interface IMigrationSettingsProvider
     [PartCreationPolicy(CreationPolicy.NonShared)]
     internal class MigrationSettingsProvider : IMigrationSettingsProvider
     {
-        private readonly IServiceProvider serviceProvider;
-        private readonly IThreadHandling threadHandling;
```

Comment:
This class doesn't need `IThreadHandling` any more. The only thread transitions that are required are handled by `ISolutionInfoProvider`.